src/lib/eslint/ESLintEngine.ts

```diff
@@ -21,74 +109,137 @@ export class ESLintEngine implements RuleEngine {
 		if (this.initialized) {
 			return;
 		}
-
 		this.logger = await Logger.child(this.getName());
 
+		this.fileHandler = new FileHandler();
+		this.config = await Controller.getConfig();
+
+		// We currently assume you must execute the engine from your project directory.  This doesn't matter for all
+		// situations, but it is critical at least for typescript.  TS rules require tsconfig.  In the future we could
+		// change to treat targetPaths which are directories as project directories, and look inside them for a
+		// tsconfig.json file.  Until then, just require that if you want to run against *.ts, you better do so from a
+		// working directory that contains tsconfig.
+		this.esconfig = await this.fileHandler.exists("tsconfig.json") ?
+			DEFAULT_ESCONFIG_TS : DEFAULT_ESCONFIG;
+
 		this.initialized = true;
 	}
 
 	getCatalog(): Promise<Catalog> {
-		return Promise.resolve({rulesets: [], categories: [], rules: []});
+		const categoryMap: Map<string, RuleGroup> = new Map();
+		const catalog: Catalog = {rulesets: [], categories: [], rules: []};
+		const rules: Rule[] = [];
+		const cli = new CLIEngine(this.esconfig);
+		cli.getRules().forEach((rule: ESRule, key: string) => {
+			const docs = rule.meta.docs;
+			const categoryName = docs.category;
+			let category = categoryMap.get(categoryName);
+			if (!category) {
+				category = {name: categoryName, engine: ESLintEngine.NAME, paths: []};
+				categoryMap.set(categoryName, category);
+			}
+			category.paths.push(docs.url);
+
+			const r = {
+				engine: ESLintEngine.NAME,
+				sourcepackage: ESLintEngine.NAME,
+				name: key,
+				description: docs.description,
+				categories: [docs.category],
+				rulesets: [docs.category],
+				languages: [ESLintEngine.JAVASCRIPT_LANGUAGE],
+				defaultEnabled: docs.recommended,
+				url: docs.url
+			};
+			if (key.startsWith("@typescript")) {
+				// Typescript is superset of javascript, so add it on top here.
+				r.languages.push(ESLintEngine.TYPESCRIPT_LANGUAGE);
+			}
+			rules.push(r)
+		});
+
+		catalog.categories = Array.from(categoryMap.values());
+		catalog.rules = rules;
+		return Promise.resolve(catalog);
 	}
 
 	public async getAll(): Promise<Rule[]> {
-		// PmdCatalogWrapper is a layer of abstraction between the commands and PMD, facilitating code reuse and other goodness.
-		this.logger.trace('Getting PMD rules.');
+		this.logger.trace('Getting eslint rules.');
 		return Promise.resolve([]);
```

Comment:
There should either be a TODO for implementing this method, or a comment explaining why it's fine as is.

Reply:
Oops, getAll() should have been removed, it isn't called from anywhere.  Done.